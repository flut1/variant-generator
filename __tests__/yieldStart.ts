import createVariants from '../src/index';

describe('createVariants()', () => {
  test('it handles yields in helper functions with yield star', () => {

    function* getFirstProp() {
      return {
        foo: yield [1, 2],
        foobar: yield ['A', 'B'],
      }
    }

    function* getSecondProp() {
      return {
        foo: yield [3, 4],
        foobar: yield ['C', 'D'],
      }
    }

    const result = createVariants(function* getConfig() {
      return {
        firstProp: yield* getFirstProp(),
        secondProp: yield* getSecondProp(),
      };
    });

    expect(Array.from(result)).toMatchSnapshot();
  });
});
