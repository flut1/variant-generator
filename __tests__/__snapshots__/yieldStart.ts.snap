// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createVariants() it handles yields in helper functions with yield star 1`] = `
Array [
  Object {
    "firstProp": Object {
      "foo": 1,
      "foobar": "A",
    },
    "secondProp": Object {
      "foo": 3,
      "foobar": "C",
    },
  },
  Object {
    "firstProp": Object {
      "foo": 1,
      "foobar": "A",
    },
    "secondProp": Object {
      "foo": 3,
      "foobar": "D",
    },
  },
  Object {
    "firstProp": Object {
      "foo": 1,
      "foobar": "A",
    },
    "secondProp": Object {
      "foo": 4,
      "foobar": "C",
    },
  },
  Object {
    "firstProp": Object {
      "foo": 1,
      "foobar": "A",
    },
    "secondProp": Object {
      "foo": 4,
      "foobar": "D",
    },
  },
  Object {
    "firstProp": Object {
      "foo": 1,
      "foobar": "B",
    },
    "secondProp": Object {
      "foo": 3,
      "foobar": "C",
    },
  },
  Object {
    "firstProp": Object {
      "foo": 1,
      "foobar": "B",
    },
    "secondProp": Object {
      "foo": 3,
      "foobar": "D",
    },
  },
  Object {
    "firstProp": Object {
      "foo": 1,
      "foobar": "B",
    },
    "secondProp": Object {
      "foo": 4,
      "foobar": "C",
    },
  },
  Object {
    "firstProp": Object {
      "foo": 1,
      "foobar": "B",
    },
    "secondProp": Object {
      "foo": 4,
      "foobar": "D",
    },
  },
  Object {
    "firstProp": Object {
      "foo": 2,
      "foobar": "A",
    },
    "secondProp": Object {
      "foo": 3,
      "foobar": "C",
    },
  },
  Object {
    "firstProp": Object {
      "foo": 2,
      "foobar": "A",
    },
    "secondProp": Object {
      "foo": 3,
      "foobar": "D",
    },
  },
  Object {
    "firstProp": Object {
      "foo": 2,
      "foobar": "A",
    },
    "secondProp": Object {
      "foo": 4,
      "foobar": "C",
    },
  },
  Object {
    "firstProp": Object {
      "foo": 2,
      "foobar": "A",
    },
    "secondProp": Object {
      "foo": 4,
      "foobar": "D",
    },
  },
  Object {
    "firstProp": Object {
      "foo": 2,
      "foobar": "B",
    },
    "secondProp": Object {
      "foo": 3,
      "foobar": "C",
    },
  },
  Object {
    "firstProp": Object {
      "foo": 2,
      "foobar": "B",
    },
    "secondProp": Object {
      "foo": 3,
      "foobar": "D",
    },
  },
  Object {
    "firstProp": Object {
      "foo": 2,
      "foobar": "B",
    },
    "secondProp": Object {
      "foo": 4,
      "foobar": "C",
    },
  },
  Object {
    "firstProp": Object {
      "foo": 2,
      "foobar": "B",
    },
    "secondProp": Object {
      "foo": 4,
      "foobar": "D",
    },
  },
]
`;
